client/post-editor/editor-sidebar/index.jsx

```diff
@@ -39,37 +45,50 @@ export default class EditorSidebar extends Component {
 			type,
 			setPostDate,
 			isPostPrivate,
+			nestedSidebar,
+			toggleNestedSidebar,
 		} = this.props;
 
+		const headerToggleSidebar = nestedSidebar === NESTED_SIDEBAR_NONE
+			? toggleSidebar
+			: partial( toggleNestedSidebar, NESTED_SIDEBAR_NONE );
```

Comment:
Same as earlier when I mentioned `partial`. Note that, in this case, the computed `headerToggleSidebar` is passed as a prop to `EditorSidebarHeader`. Since any function returned from `partial` will always have a different reference, now the child component has no way to know, just by looking at props' references, whether it should update.